### PR TITLE
[IMP] hr, hr_holidays : improve department's follower

### PR DESCRIPTION
**Pull request by mepe-odoo**
When the department's manager changes, he will unfollow the department. Follower's subtypes introduced by hr_holidays become optional

task-3817606

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr/models/hr_department.py [hunk 1]
```diff
@@ -99,6 +99,8 @@ def write(self, vals):
                 # subscribe the manager user
                 if manager.user_id:
                     self.message_subscribe(partner_ids=manager.user_id.partner_id.ids)
+            if self.manager_id and self.manager_id.user_id:
```
Comment (tivisse) [inline addons/hr/models/hr_department.py:102]: You should filter the records for which the manager actually changed. Otherwise you'll subscribe him anyway.

```suggestion
            for department in self:
                if department.manager_id.user_id and department.manager_id != manager:
                    department.message_unsubscribe(partner_ids=department.manager_id.user_id.partner_id.ids)
```

File: addons/hr_holidays/models/mail_message_subtype.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def _update_department_subtype(self):
                 department_subtype = self.create({
                     'name': subtype.name,
                     'res_model': 'hr.department',
-                    'default': subtype.default or False,
```
Comment (tivisse) [inline addons/hr_holidays/models/mail_message_subtype.py:31]: Why this change ?

**Review by tivisse — CHANGES_REQUESTED** [review]
Small comment.

File: addons/hr_holidays/models/mail_message_subtype.py (hunk 2 above)
Comment (mepe-odoo) [reply addons/hr_holidays/models/mail_message_subtype.py:31]: Because Geoffrey don't want to have this type subscribed by defaut :)

File: addons/hr/models/hr_department.py (hunk 1 above)
Comment (mepe-odoo) [reply addons/hr/models/hr_department.py:102]: yep yep I will corretc it

**Comment by tivisse** [comment]
@robodoo r+
